pkg/scheduler/objects/application_state.go

```diff
@@ -172,6 +175,9 @@ func NewAppState() *fsm.FSM {
 			},
 			fmt.Sprintf("leave_%s", Running.String()): func(event *fsm.Event) {
 				app := event.Args[0].(*Application) //nolint:errcheck
+				if app.queue != nil {
```

Comment:
The app must have a queue set if it is in running state, there is no need for a nil check